src/core/views.py

```diff
@@ -3103,10 +3090,18 @@ def view_proposal_review_decision(
                     )
                     notification.save()
 
-            if access_key:
-                return redirect(reverse('proposal_review_declined'))
-            else:
-                return redirect(reverse('reviewer_dashboard'))
+        redirect_kwargs = {
+            'proposal_id': _proposal.id,
+            'assignment_id': assignment_id,
+            'decision': decision,
```

Comment:
if both `'accept'` and `'decline'` are not in `request.POST`, `decision` won't have a value and this will crash. 
Is there a chance of this happening?

Reply:
Yeah, well spotted. I was attempting to reduce a it of repetition, but yes the `redirect` call(s) should only be made in logic paths where `accept` or `decline` are in `decision.POST`.